File: backend/EcommerceApi/Models/Pedido.cs
namespace EcommerceApi.Models
{
    public class Pedido
    {
        public int Id { get; set; }

        public DateTime DataCriacao { get; set; } = DateTime.Now;

        public List<ItemPedido> Itens { get; set; } = new List<ItemPedido>();

        public decimal Total => Itens.Sum(item => item.Quantidade * item.PrecoUnitario);
    }
}

File: frontend/ecommerce-app/src/app/core/models/pedido.ts
export interface Pedido {
    id: number;
    dataCriacao: Date;
    itens: itemPedidoDto[];
    total: number;
}

export interface itemPedidoDto{
    idProduto: number;
    idPedido: number;
    produto: Produtos;
    quantidade: number;
    precoUnitario: number;
}

export interface Produtos{
    id: number;
    nome: string;
    preco: number;
}